File: java/spring/spring-switch-controller/src/main/java/sample/SampleController.java
package sample;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
public class SampleController {

    @RequestMapping("/sample/{id}")
    public String sample(@PathVariable("id") String id, Model model) {
        System.out.println("#### sample : " + id);

        model.addAttribute("data", "sample:" + id);
        return "sample";
    }
}

File: kotlin/spring/webflux/mongodb-sample/src/main/kotlin/sample/controller/SampleController.kt
package sample.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import sample.data.Data
import sample.repository.DataRepository

@RestController
class SampleController {
    @Autowired
    lateinit var dataRepository: DataRepository

    @GetMapping("/data/{id}")
    fun getData(@PathVariable("id") id: String) = dataRepository.findData(id)

    @PutMapping("/data")
    fun putData(@RequestBody data: Data) = dataRepository.saveData(data)
}